src/Raven.Server/Documents/ConflictsStorage.cs

```diff
@@ -815,37 +816,62 @@ private static int FindIndexOfLargestEtagAndMergeChangeVectors(IReadOnlyList<Doc
             return indexOfLargestEtag;
         }
 
-        public static ConflictStatus GetConflictStatusForDocument(DocumentsOperationContext context, string id, LazyStringValue remote, string priority, out string conflictingVector)
+        public static ConflictStatus GetConflictStatusForDocument(DocumentsOperationContext context, IncomingReplicationHandler.ReplicationItem remote,
+            out string conflictingVector, out bool removeClusterTx)
         {
+
+            var hasRemoteClusterTx = remote.Flags.Contain(DocumentFlags.FromClusterTransaction);
+            bool hasLocalClusterTx;
+            removeClusterTx = false;
+
             //tombstones also can be a conflict entry
             conflictingVector = null;
-            var conflicts = context.DocumentDatabase.DocumentsStorage.ConflictsStorage.GetConflictsFor(context, id);
+            var conflicts = context.DocumentDatabase.DocumentsStorage.ConflictsStorage.GetConflictsFor(context, remote.Id);
+            ConflictStatus status;
             if (conflicts.Count > 0)
             {
+                // an incoming document from a cluster transaction will always resolve any local _existing_ conflicts
+                if (hasRemoteClusterTx)
+                    return ConflictStatus.Update;
+
                 foreach (var existingConflict in conflicts)
                 {
-                    if (ChangeVectorUtils.GetConflictStatus(remote, existingConflict.ChangeVector, priority) == ConflictStatus.Conflict)
+                    status = ChangeVectorUtils.GetConflictStatus(remote.ChangeVector, existingConflict.ChangeVector);
+                    if (status == ConflictStatus.Conflict)
                     {
                         conflictingVector = existingConflict.ChangeVector;
-                        return ConflictStatus.Conflict;
+                        return status;
                     }
                 }
                 // this document will resolve the conflicts when putted
                 return ConflictStatus.Update;
             }
 
-            var result = context.DocumentDatabase.DocumentsStorage.GetDocumentOrTombstone(context, id);
+            var result = context.DocumentDatabase.DocumentsStorage.GetDocumentOrTombstone(context, remote.Id);
             string local;
 
             if (result.Document != null)
+            {
                 local = result.Document.ChangeVector;
+                hasLocalClusterTx = result.Document.Flags.Contain(DocumentFlags.FromClusterTransaction);
+            }
             else if (result.Tombstone != null)
+            {
                 local = result.Tombstone.ChangeVector;
+                hasLocalClusterTx = result.Tombstone.Flags.Contain(DocumentFlags.FromClusterTransaction);
+            }
             else
                 return ConflictStatus.Update; //document with 'id' doesn't exist locally, so just do PUT
 
+            if (hasLocalClusterTx || hasRemoteClusterTx)
```

Comment:
This might be too early

Reply:
This is only to mark that we need to remove the ClusterTx flag when we will put the document.

Reply:
Needs to be moved _after_ you know that this isn't already okay